kinto/core/utils.py

```diff
@@ -443,3 +444,40 @@ def instance_uri(request, resource_name, **params):
     """Return the URI for the given resource."""
     return strip_uri_prefix(request.route_path('%s-record' % resource_name,
                                                **params))
+
+
+def apply_json_patch(record, ops):
+    """
+    Apply JSON Patch operations using jsonpatch.
+
+    :param record: base record where changes should be applied (not in-place).
+    :param list changes: list of JSON patch operations.
+    :param bool only_data: param to limit the scope of the patch only to 'data'.
+    :returns dict data: patched record data.
+             dict permissions: patched record permissions
+    """
+    data = record.copy()
+    permissions = {'read': set(), 'write': set()}
+
+    # To be used with SharableResource object (patching permissions)
+    if '__permissions__' in record:
+        permissions.update(record['__permissions__'])
+
+    # Map permissions as dicts since jsonpatch module doesn't accept sets.
+    permissions = {k: {i: i for i in v} for k, v in permissions.items()}
```

Comment:
Oh, very interesting! Please add all those comments regarding removing by value along the code :)